ocean_lib/models/bpool.py

```diff
@@ -426,15 +426,9 @@ def exitswapExternAmountOut(
     def totalSupply(self) -> int:
         return self._ccontract.totalSupply()
 
-    def balanceOf(self, whom_address: str) -> int:
```

Comment:
Why is `balanceOf()` and `approve()` removed? 

Notes: BPool class has ERC20 functionality, starting at the line `# ==== Balancer Pool as ERC20`. I implemented them here rather than relying on its parent class BToken's implementation, on purpose.

Reply:
It was the exact same code as in the parent class, only different variable names, so I removed the lines. Should I add them back?

Reply:
It's not the exact same code: they have different variable names for the concise contract. `_ccontract` vs `contract_concise`.

Yes, add them back please.

There is probably a small opportunity for cleanup here. But it's a bit dangerous, so I recommend keeping them apart for now. This code has been pretty rock solid with few changes since release, lots of unit tests, and quite clean, so I don't see much need to fiddle with it.

One final thought: the unit tests should have picked up an error after you deleted the lines. Did they not? If no, it points to that we need better tests yet. (And we know that! Right now even though this is the most thorough testing in all of ocean.py, they're still only mostly happy-path tests.)